lib/mocha.js

```diff
@@ -277,15 +283,67 @@ Mocha.prototype.invert = function() {
  * @param {Boolean} ignore
  * @return {Mocha}
  * @api public
- * @param {boolean} ignore
- * @return {Mocha}
  */
 Mocha.prototype.ignoreLeaks = function(ignore) {
   this.options.ignoreLeaks = Boolean(ignore);
   return this;
 };
 
 /**
+ * Enable or disable randomization of test execution order within suites.
+ *
+ * @param {(boolean|number|string)} [seed] Optional random seed.  Seed must be
+ *   a 32-bit unsigned integer, or a string which can convert to one.  If
+ *   `true`, a seed will be created.  If `false`, randomization will be
+ *   globally disabled (if it was enabled previously).
+ * @return {Mocha}
+ * @api public
+ */
+Mocha.prototype.randomize = function randomize(seed) {
+  if (seed !== false) {
+    if (!arguments.length || seed === true) {
+      // generate a number for seeding
+      seed = Mocha.seed();
+    } else if (!(typeof seed === 'number' && seed > 0
+      && seed <= MAX_SAFE_INTEGER && seed % 1 === 0)) {
+      // this would be a hexadecimal number represented as a string
+      if (/^(0[xX])?[a-fA-F0-9]{1,14}$'/.test(seed)) {
+        seed = parseInt(seed, 16);
+      } else {
+        // a decimal number as string, presumably
+        seed = parseInt(seed, 10);
+        if (isNaN(seed)) {
+          throw new Error('Invalid random seed. Expected unsigned 32-bit '
+            + 'integer; got: "' + seed + '"');
+        }
+      }
+    }
+
+    var engine = Random.engines.mt19937().seed(seed);
+    this.options.randomConfig = {
+      shuffleTests: function shuffleTests(tests) {
+        Random.shuffle(tests, engine);
+        return tests;
+      },
+      seed: seed,
+      hex: '0x' + Number(seed).toString(16) // for display
+    };
+  } else {
+    delete this.options.randomConfig;
+  }
+  return this;
+};
+
+/**
+ * Generate a random seed.
+ * @returns {number} Unsigned 32-bit integer
+ */
+Mocha.seed = function seed() {
+  return Random.integer(0, MAX_SAFE_INTEGER)(Random.engines.mt19937()
+    .autoSeed());
+};
```

Comment:
That comment would be something like "go read the docs for this other library".  😜 

Basically, you can't get a value out of the engine unless it's seeded.  So if we want a seed, then we should randomly generate one.  And if we're randomly generating a seed, why not use the random number generator..?